arithmetization/src/test/java/net/consensys/linea/zktracer/exceptions/OutOfGasMemExpExceptionTest.java

```diff
@@ -0,0 +1,604 @@
+
+    bytecodeRunner.run(gasCost + cornerCase, List.of(codeOwnerAccount));
+    if (cornerCase == -1) {
+      assertEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    } else {
+      assertNotEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {-1, 0, 1})
+  void outOfGasExceptionRevert(int cornerCase) {
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    program
+        .push(
+            Bytes.fromHexString(
+                "0xF00000000000000000000000000000000000000000000000000000000000FFAA")) // value
+        .push(0) // offset
+        .op(OpCode.MSTORE)
+        .push(3) // value
+        .push(30) // offset
+        .op(OpCode.REVERT);
+
+    Bytes pgCompile = program.compile();
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(pgCompile);
+
+    long gasCost = bytecodeRunner.runOnlyForGasCost();
+
+    bytecodeRunner.run(gasCost + cornerCase);
+    if (cornerCase == -1) {
+      assertEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    } else {
+      assertNotEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {-6420, -6419, -6418, 100, 101, 102})
+  /*
+  Deployment code: "0x7F7EFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF60005260206000F3"
+  1. OOGX for CREATE before deployment: remove 6400 (depositFee) + deployment code exec cost (18)
+  2. OOGX for CREATE after deployment: enough gas for child creation, but not enough to complete deployment code or deposit
+  3. No OOGX for CREATE after deployment: add 1/64th of 6418 to gasCost to account for gasAvailableForChildCreate
+   */
+  void outOfGasExceptionCreate(int cornerCase) {
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    program
+        // constructor
+        .push(
+            Bytes.fromHexString(
+                "0x7F7EFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")) // value
+        .push(0) // offset
+        .op(OpCode.MSTORE)
+        .push(
+            Bytes.fromHexString(
+                "0xFF60005260206000F30000000000000000000000000000000000000000000000")) // value
+        .push(32) // offset
+        .op(OpCode.MSTORE)
+        // Create the contract
+        .push(41)
+        .push(0)
+        .push(0)
+        .op(OpCode.CREATE); // No constructor so code executed and runtime code set to return value
+
+    Bytes pgCompile = program.compile();
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(pgCompile);
+
+    long gasCost = bytecodeRunner.runOnlyForGasCost();
```

Comment:
Are we sure this is the correct `gasCost` ? I'm asking because a naïve implementation of `runOnlyForGasCost` would compute the total gas consumed by the transaction, say. But this would include the `PUSH`'s and `MSTORE`'s and `RETURN`'s of the deployment inside the `CREATE`. Are we sure those costs aren't taken into account when running `runOnlyForGasCost` ?

What we need here is the cost just up to an including the `upfrontGasCost` of the `CREATE`.

Reply:
Oh, I guess this is what the `6420` etc is about ...